modules/tensor_mechanics/include/kernels/ADInertialForceShell.h

```diff
@@ -0,0 +1,186 @@
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _old_vel_0, _old_vel_1, _old_rot_vel_0, _old_rot_vel_1;
+  ADRealVectorValue _old_vel_2, _old_vel_3, _old_rot_vel_2, _old_rot_vel_3;
+  /**
+   * Current translational and rotational accelerations at the two nodes
+   * of the beam in the global coordinate system
+   **/
+  ADRealVectorValue _accel_0, _accel_1, _rot_accel_0, _rot_accel_1;
+  ADRealVectorValue _accel_2, _accel_3, _rot_accel_2, _rot_accel_3;
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_vel_0, _local_vel_1, _local_rot_vel_0, _local_rot_vel_1;
+  ADRealVectorValue _local_vel_2, _local_vel_3, _local_rot_vel_2, _local_rot_vel_3;
+  /**
+   * Current translational and rotational velocities at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_old_vel_0, _local_old_vel_1, _local_old_rot_vel_0, _local_old_rot_vel_1;
+  ADRealVectorValue _local_old_vel_2, _local_old_vel_3, _local_old_rot_vel_2, _local_old_rot_vel_3;
+  /**
+   * Current translational and rotational accelerations at the two nodes
+   * of the beam in the initial beam local coordinate system
+   **/
+  ADRealVectorValue _local_accel_0, _local_accel_1, _local_rot_accel_0, _local_rot_accel_1;
+  ADRealVectorValue _local_accel_2, _local_accel_3, _local_rot_accel_2, _local_rot_accel_3;
+  /**
+   * Forces and moments at the two end nodes of the beam in the initial
+   * beam local configuration
+   **/
+  std::vector<ADRealVectorValue> _local_force, _local_moment;
+
+  /**
+   * Forces and moments at the two end nodes of the beam in the global
+   * coordinate system
+   **/
+  ADRealVectorValue _global_force_0, _global_force_1, _global_moment_0, _global_moment_1;
+  ADRealVectorValue _global_force_2, _global_force_3, _global_moment_2, _global_moment_3;
+
+  // AMR
+  /// Derivatives of shape functions w.r.t isoparametric coordinates xi
+  std::vector<std::vector<Real>> _dphidxi_map;
+
+  /// Derivatives of shape functions w.r.t isoparametric coordinates eta
+  std::vector<std::vector<Real>> _dphideta_map;
+
+  /// Shape function value
+  std::vector<std::vector<Real>> _phi_map;
+
+  /// Vector storing pointers to the nodes of the shell element
+  std::vector<const Node *> _nodes;
+
+  /// Quadrature points in the in-plane direction in isoparametric coordinate system
+  std::vector<Point> _2d_points;
+
+  /// Quadrature weights
+  std::vector<ADReal> _2d_weights;
+
+  /// First tangential vectors at nodes
+  std::vector<ADRealVectorValue> _v1;
+
+  /// Second tangential vectors at nodes
+  std::vector<ADRealVectorValue> _v2;
+
+  /// Helper vectors
+  ADRealVectorValue _x2;
+  ADRealVectorValue _x3;
+
+  /// Normal to the element at the 4 nodes.
+  std::vector<ADRealVectorValue> _node_normal;
+
+  /// Node 1 g vector in reference configuration (mass matrix)
```

Comment:
this should be a single `std::array<ADRealVectorValue, 4> _0g_vector;` (not the one-off indexing)

Reply:
There are two vectors at each node.  I kept them separate but made them an array:

`std::array<ADRealVectorValue, 2> _0g1_vectors` 

The one-off indexing is just meant to preserve the notation from papers/PhD thesis. Rest of the comments should be fully addressed.